src/home/homectl.c

```diff
@@ -2474,6 +2475,42 @@ static int create_interactively(void) {
         if (r < 0)
                 return log_error_errno(r, "Failed to set userName field: %m");
 
+        for (;;) {
```

Comment:
hmm, I think we really should show a list of available groups first. i.e. enumerate all groups with groupdb_all(), suppress entries that do not have a disposition of either USER_SYSTEM or USER_REGULAR.

And then maybe finally suppress all groups for which identically named users exist and where both dispositions are USER_REGULAR.

Those restrictions should just be made when listing suggested groups. If user types in anything not listed we should just accept that.

Reply:
maybe generalize the show_menu() code from firstboot.c